### DISCUSSION TITLE
How do I install ripgrep on Linux if I don't have root permissions?

### DISCUSSION THREAD

**Discussion by luo4lu** [BurntSushi/ripgrep#2477] [Q&A] [answered]
#### Describe your feature request
I want to install this tool only for current user. Because I don't have root permission.

**Comment by BurntSushi** [comment, marked as answer]
The very beginning of the [installation instructions in the README](https://github.com/BurntSushi/ripgrep#installation) answer this:

> [Archives of precompiled binaries for ripgrep are available for Windows, macOS and Linux.](https://github.com/BurntSushi/ripgrep/releases) Linux and Windows binaries are static executables. Users of platforms not explicitly mentioned below are advised to download one of these archives.

It doesn't specifically call out that you don't need root for this, but it's just downloading some files. You shouldn't need root for that. So for example, to download and use ripgrep without root using pre-compiled binaries (this should work on any x86-64 Linux):

```
$ curl -LO 'https://github.com/BurntSushi/ripgrep/releases/download/13.0.0/ripgrep-13.0.0-x86_64-unknown-linux-musl.tar.gz'
$ tar xf ripgrep-13.0.0-x86_64-unknown-linux-musl.tar.gz
$ cd ripgrep-13.0.0-x86_64-unknown-linux-musl
$ ./rg --version
ripgrep 13.0.0 (rev af6b6c543b)
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

You can then put the `rg` binary anywhere. Convention might be `$HOME/bin`, but usually any directory that's in your `PATH` is suitable.
